### DISCUSSION TITLE
Best Practice on Supabase

### DISCUSSION THREAD

**Discussion by Tutuldot** [supabase/supabase#13599] [Questions]
Hi, 

I am new to Supabase and restapi. i have question regarding on the best practices on inserting or updating data. When i need to insert data to 2 tables (parent and child table 1 to many relationship), is it ok to have separate post request for insertion to parent then to child table? or is it better to create a postgres function to insert both using the function?

Thanks 

Anthony

**Comment by GaryAustin1** [comment]
If you do it from the client you need to make sure to handle error of 2nd request and back out first. 
if you use the function, both operations will run as a transaction so if either fails they both fail.

**Reply by Tutuldot** [reply]
thank you
